### PR TITLE
Remove custom JDBC driver

### PR DESCRIPTION
**Pull request by bobbyiliev**
As part of the initial work for the Metabase Materialize partner driver, this PR removes the old custom JDBC connector which was required in the past.

metabase-materialize-driver

### CODE REVIEW

**Review by benesch — APPROVED** [review]

**Review by dehume — APPROVED** [review]

File: resources/metabase-plugin.yaml [hunk 1]
```diff
@@ -1,35 +1,33 @@
...
-        - default: materialize
-    - merge:
-        - user
-        - default: default
-    - merge:
-        - password
-        - default: default
-  connection-properties-include-tunnel-config: false
+    - name: dbname
+      display-name: Databases
```
Comment (joacoc) [inline resources/metabase-plugin.yaml:20]:
```suggestion
      display-name: Database
```

File: resources/metabase-plugin.yaml [hunk 2]
```diff
@@ -1,35 +1,33 @@
...
+      helper-text: Your Materialize user name.
+    - password
+    - ssl
 init:
   - step: load-namespace
     namespace: metabase.driver.materialize
   - step: register-jdbc-driver
-    class: io.materialize.Driver
+    class: com.materialize.jdbc.MaterializeDriver
+    # class: org.postgresql.Driver
```
Comment (joacoc) [inline resources/metabase-plugin.yaml:33]:
```suggestion
```

**Review by joacoc — COMMENTED** [review]
Two subtle details.

File: resources/metabase-plugin.yaml (hunk 1 above)
Comment (joacoc) [reply resources/metabase-plugin.yaml:20]: Unless you can use multiple databases separated by commas

**Pull request merged**